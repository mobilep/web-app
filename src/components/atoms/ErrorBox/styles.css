.ErrorBox {
	align-items: center;
	background: var(--white);
	border-radius: 5px;
	color: var(--tomatoRed);
	display: flex;
	font-size: 14px;
	padding: 10px;
}

.ErrorBox-withIcon {
	padding-left: 15px;
}

.ErrorBox-withIcon .ErrorBox-message {
	margin-left: 10px;
}
