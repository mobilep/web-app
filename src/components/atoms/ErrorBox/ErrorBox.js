import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './styles.css';

export default class ErrorBox extends Component {

	getClassName ({ icon, className }) {
		return classNames({
			ErrorBox: true,
			'ErrorBox-withIcon': icon,
		}, className);
	}

	render () {
		const className = this.getClassName(this.props);
		const { icon = '', message } = this.props;
		return (
			<div className={className}>
				{icon}<span className='ErrorBox-message'>{message}</span>
			</div>
		);
	}
}

ErrorBox.propTypes = {
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.array]),
	className: PropTypes.string,
	icon: PropTypes.element,
	message: PropTypes.string,
};
