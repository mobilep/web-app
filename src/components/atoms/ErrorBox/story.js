import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';
import ErrorBox from './ErrorBox';

storiesOf('1. Atoms / ErrorBox', module)
	.add('ErrorBox', () => {
		const message = text('error text', `Uh no, we don't recognise that email address`);
		return (
			<ErrorBox {...{ message }} />
		);
	});
