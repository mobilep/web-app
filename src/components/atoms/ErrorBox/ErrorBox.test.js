import React from 'react';
import { shallow } from 'enzyme';

import ErrorBox from './ErrorBox';

describe('Atom - ErrorBox', () => {

	it('Should render', () => {
		const wrapper = shallow(<ErrorBox message='Message' />);
		expect(wrapper).toMatchSnapshot();
	});

});

