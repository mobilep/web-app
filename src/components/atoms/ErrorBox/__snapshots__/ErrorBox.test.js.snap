// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - ErrorBox Should render 1`] = `
<div
  className="ErrorBox"
>
  <span
    className="ErrorBox-message"
  >
    Message
  </span>
</div>
`;
